src/containers/GraphSettingsContainer.tsx

```diff
@@ -64,6 +75,15 @@ class GraphSettings extends React.PureComponent<GraphSettingsProps> {
     });
   };
 
+  handleToggleServiceNodes = () => {
+    const currentShowServiceNodes = this.props.showServiceNodes;
+    this.props.toggleServiceNodes(); // this is async so props are not immediately updated
+    this.handleFilterChangeToUrl({
+      ...this.getGraphParams(),
+      injectServiceNodes: !currentShowServiceNodes
```

Comment:
I think you should use `componendDidUpdate` instead to reflect this change.
```
componentDidUpdate() {
    this.handleFilterChangeToUrl({... this.getGraphParams(), injectServiceNodes: this.props.showServiceNodes}); 
  }
```

but having `injectServiceNodes: this.props.showServiceNodes` seems rare.

Reply:
@josejulio OK, that was a good idea, thanks.  The life-cycle function is a more natural way to handle the control flow.  And it simplifies things not needing the additional handler function.